### PR TITLE
@broskoski: Lots of little QA

### PR DESCRIPTION
**Pull request by craigspaeth**
- Validates full email regex
- Leaving guest fields blank pops validation (replaced controller strip logic with model validation + disable/enable inputs through JS)
- Scoped uniqueness of email validation to per-event
- Added closing date field
- Refactored `waitlisted` to a method that dynamically queries if the RSVP's index among the ordered RSVP list is above capacity
- Export to CSV formats the guests array

### CODE REVIEW

**Comment by craigspaeth** [comment]
Gonna marge, but lmk if you have any comments.

File: app/assets/javascripts/rsvps.coffee [hunk 1]
```diff
@@ -6,4 +6,6 @@ $ ->
   $('#rsvp_no_of_guests').on 'change', (e) ->
     number = e.target.value
     $(".rsvp__guests  .field").hide()
-    $(".rsvp__guests  .field:lt(#{number})").show()
\ No newline at end of file
+      .find('input').prop('disabled', true)
```
Comment (broskoski) [inline app/assets/javascripts/rsvps.coffee:9]: Nice

File: app/models/rsvp.rb [hunk 2]
```diff
@@ -7,22 +7,29 @@ class Rsvp < ApplicationRecord
...
   validates :email,
     presence: { message: "must be present" },
-    format: { with: /@/, message: "address must be valid"},
-    uniqueness: { message: "has already RSVPed"}
+    format: {
+      with: /\A[^@\s]+@([^@\s]+\.)+[^@\s]+\z/,
+      message: "address must be valid"
+    },
+    uniqueness: { scope: :event, message: "has already RSVPed"}
+  validate :guests_cannot_be_blank
```
Comment (broskoski) [inline app/models/rsvp.rb:15]: Nice

File: app/models/rsvp.rb [hunk 3]
```diff
@@ -7,22 +7,29 @@ class Rsvp < ApplicationRecord
...
+    index + 1 > event.capacity
   end
 
   def self.to_csv(options = {})
     CSV.generate(options) do |csv|
       csv << column_names
       all.each do |rsvp|
-        csv << rsvp.attributes.values
+        attrs = rsvp.attributes
+        attrs["guests"] = attrs["guests"].join ', '
```
Comment (broskoski) [inline app/models/rsvp.rb:31]: ✅

**Comment by broskoski** [comment]
Thank ye!

**Pull request merged**